pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -576,15 +576,34 @@ def available(self, exception_flag=True):
         """True if the solver is available."""
         exe = pyomo.common.Executable("gams")
         if exception_flag is False:
-            return exe.available()
+            if not exe.available():
+                return False
         else:
-            if exe.available():
-                return True
-            else:
+            if not exe.available():
                 raise NameError(
                     "No 'gams' command found on system PATH - GAMS shell "
                     "solver functionality is not available.")
 
+        try:
+            from gdxcc import new_gdxHandle_tp, gdxCreateD, gdxClose, gdxFree
+            from gdxcc import gdxOpenRead, gdxDataReadRawStart, gdxDataReadRaw
+            from gdxcc import gdxDataReadDone, gdxSymbolInfo
+            return True
+        except ImportError as e:
+            if not exception_flag:
+                return False
+            else:
+                raise ImportError("Import of gams failed - GAMS direct "
+                                  "solver functionality is not available.\n"
+                                  "GAMS message: %s" % (e,))
+        except:
+            logger.warning(
+                "Attempting to import gams generated unexpected exception:\n"
+                "\t%s: %s" % (sys.exc_info()[0].__name__, sys.exc_info()[1]))
+            if not exception_flag:
+                return False
+            raise
```

Comment:
We might want to use the deferred import system for this. @jsiirola do you have documentation somewhere for that?

Reply:
I think it is "6 of one, half dozen of the other".  If you wanted to use the deferred import, it would look something like:
```python
from pyomo.common.dependencies import attempt_import
gdxcc, gdxcc_available = attempt_import('gdxcc', defer_check=True)
```
Then `available()` just needs to use:
```python
if gdxcc_available:
    return True
elif exception_flag:
    raise ImportError("Import of gams failed - GAMS direct "
                      "solver functionality is not available.\n"
                      "GAMS message: %s" % (e,))
else:
    return False
```